From 642d21c486e42afe5b897027014ed5812250b498 Mon Sep 17 00:00:00 2001
From: Arturo Moya <0x616d@gmail.com>
Date: Fri, 4 Sep 2026 02:59:55 -0400
Subject: [PATCH 1/2] Fix AttributeError module 'ast' has no attribute 'Str' in python 3.14

ast.Str have been deprecated alias of Constant since Python 3.8 and have
emitted deprecation warnings since Python 3.12 and removed in Python 3.14
https://docs.python.org/dev/whatsnew/3.14.html#id9
---
 sandbox/grist/formula_prompt.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sandbox/grist/formula_prompt.py b/sandbox/grist/formula_prompt.py
index c252066..50c1140 100644
--- a/sandbox/grist/formula_prompt.py
+++ b/sandbox/grist/formula_prompt.py
@@ -205,7 +205,8 @@ def convert_completion(completion):
     if (
       len(func_body_stmts) > 1 and
       isinstance(func_body_stmts[0], ast.Expr) and
-      isinstance(func_body_stmts[0].value, ast.Str)
+      isinstance(func_body_stmts[0].value, ast.Constant) and
+      isinstance(func_body_stmts[0].value.value, str)
     ):
       # Skip the docstring.
       first_stmt = func_body_stmts[1]
-- 
2.54.0


From cbbdb9a328aaaad3da0867bff2fdbad65cc807d8 Mon Sep 17 00:00:00 2001
From: Arturo Moya <0x616d@gmail.com>
Date: Fri, 4 Sep 2026 03:13:54 -0400
Subject: [PATCH 2/2] Regenerate requirements.txt

---
 sandbox/requirements.txt | 42 +++++++++++++++++-----------------------
 1 file changed, 18 insertions(+), 24 deletions(-)

diff --git a/sandbox/requirements.txt b/sandbox/requirements.txt
index 7f8cf2e..affdad7 100644
--- a/sandbox/requirements.txt
+++ b/sandbox/requirements.txt
@@ -1,57 +1,51 @@
 #
-# This file is autogenerated by pip-compile with Python 3.9
+# This file is autogenerated by pip-compile with Python 3.14
 # by the following command:
 #
 #    pip-compile --output-file=core/sandbox/requirements.txt core/sandbox/requirements.in
 #
-astroid==2.14.2
+astroid==4.3.1
     # via -r core/sandbox/requirements.in
-asttokens==2.4.0
+asttokens==3.0.2
     # via
     #   friendly-traceback
     #   stack-data
-chardet==5.1.0
+chardet==7.6.0
     # via -r core/sandbox/requirements.in
-et-xmlfile==1.0.1
+et-xmlfile==2.0.0
     # via openpyxl
 executing==2.2.1
     # via
     #   -r core/sandbox/requirements.in
     #   friendly-traceback
     #   stack-data
-friendly-traceback==0.7.48
+friendly-traceback==0.7.61
     # via -r core/sandbox/requirements.in
-iso8601==0.1.12
+iso8601==2.1.0
     # via -r core/sandbox/requirements.in
-lazy-object-proxy==1.12.0
-    # via astroid
-openpyxl==3.0.10
+lxml==6.1.3
+    # via unittest-xml-reporting
+openpyxl==3.1.5
     # via -r core/sandbox/requirements.in
-phonenumberslite==8.12.57
+phonenumberslite==9.0.38
     # via -r core/sandbox/requirements.in
-pure-eval==0.2.2
+pure-eval==0.2.3
     # via
     #   friendly-traceback
     #   stack-data
-python-dateutil==2.8.2
+python-dateutil==2.9.0.post0
     # via -r core/sandbox/requirements.in
-roman==3.3
+roman==5.2
     # via -r core/sandbox/requirements.in
 six==1.17.0
     # via
-    #   asttokens
     #   friendly-traceback
     #   python-dateutil
-    #   unittest-xml-reporting
 sortedcontainers==2.4.0
     # via -r core/sandbox/requirements.in
-stack-data==0.5.1
+stack-data==0.6.3
     # via friendly-traceback
-typing-extensions==4.4.0
-    # via
-    #   -r core/sandbox/requirements.in
-    #   astroid
-unittest-xml-reporting==2.0.0
+typing-extensions==4.16.0
+    # via -r core/sandbox/requirements.in
+unittest-xml-reporting==4.0.0
     # via -r core/sandbox/requirements.in
-wrapt==1.15.0
-    # via astroid
-- 
2.54.0
